lib/Runtime/Library/JavascriptBigInt.cpp

```diff
@@ -397,4 +414,183 @@ namespace Js
         return (leftBigInt->Compare(rightBigInt) == 0);
     }
 
+    // pbi1 += pbi2 assume pbi1 > pbi2
+    void JavascriptBigInt::AddAbsolute(JavascriptBigInt * pbi1, JavascriptBigInt * pbi2)
+    {
+        digit_t carryDigit = 0;
+        digit_t *pDigit1 = pbi1->m_digits;
+        digit_t *pDigit2 = pbi2->m_digits;
+        digit_t i = 0;
+
+        for (; i < pbi2->m_length; i++)
+        {
+            digit_t tempCarryDigit = 0;
+            if (pDigit2[i] > 0)
+            {
+                pDigit1[i] = JavascriptBigInt::AddDigit(pDigit1[i], pDigit2[i], &tempCarryDigit);
+            }
+            if (carryDigit > 0)
+            {
+                pDigit1[i] = JavascriptBigInt::AddDigit(pDigit1[i], carryDigit, &tempCarryDigit);
+            }
+            carryDigit = tempCarryDigit;
+        }
+
+        for (; i < pbi1->m_length; i++)
+        {
+            digit_t tempCarryDigit = 0;
+            if (carryDigit > 0)
+            {
+                pDigit1[i] = JavascriptBigInt::AddDigit(pDigit1[i], carryDigit, &tempCarryDigit);
+            }
+            carryDigit = tempCarryDigit;
+        }
+
+        if (0 < carryDigit) // length increase by 1
+        {
+            if (pbi1->m_length >= pbi1->m_maxLength && !pbi1->Resize(pbi1->m_length + 1))
+            {
+                return;
```

Comment:
I see a couple of different behaviors in this file for how we handle failure to resize. I don't see this case included in the spec (it usually ignores anything to do with the reality of running on a physical machine with less than infinite memory). I could imagine three possible options:

* Return a wrong answer (what we do here)
* Throw an exception
* Crash the program (what we do on line 217)

I'd like to propose that we update Resize to return void and throw a `RangeError` (`JavascriptError::ThrowRangeError`) if it can't resize, since this seems most consistent with other parts of the spec, such as the behaviors when a program tries to create a string or array that is longer than allowed.

Reply:
will do